core/page-assets/data-managers/base.php

```diff
@@ -0,0 +1,314 @@
+	 * Getting a file content or size.
+	 *
+	 * @since 3.3.0
+	 * @access protected
+	 *
+	 * @param string $data_type (content|size)
+	 * @param string $file_key - In case that the same file data is needed for multiple assets (like a JSON file), the file data key should be the same for all shared assets to make sure that the file is being read only once.
+	 *
+	 * @return string|number
+	 */
+	protected function get_file_data( $data_type, $file_key = '' ) {
+		$asset_key = $file_key ? $file_key : $this->get_key();
+
+		if ( isset( $this->files_data[ $asset_key ][ $data_type ] ) ) {
+			return $this->files_data[ $asset_key ][ $data_type ];
+		}
+
+		if ( ! isset( $this->files_data[ $asset_key ] ) ) {
+			$this->files_data[ $asset_key ] = [];
+		}
+
+		$asset_path = $this->get_file_path();
+
+		if ( 'content' === $data_type ) {
+			$data = file_get_contents( $asset_path );
+
+			if ( ! $data ) {
+				$data = '';
+			}
+		} elseif ( 'size' === $data_type ) {
+			$data = file_exists( $asset_path ) ? filesize( $asset_path ) : 0;
+		}
+
+		$this->files_data[ $asset_key ][ $data_type ] = $data;
+
+		return $data;
+	}
+
+	/**
+	 * Get Saved Assets Data.
+	 *
+	 * Getting the assets data from the DB.
+	 *
+	 * @since 3.3.0
+	 * @access protected
+	 *
+	 * @return array
+	 */
+	protected function get_saved_assets_data() {
+		return get_option( self::ASSETS_DATA_KEY, [] );
+	}
+
+	/**
+	 * Init Asset Data.
+	 *
+	 * Initialize the asset data and handles the asset content updates when needed.
+	 *
+	 * @since 3.3.0
+	 * @access public
+	 *
+	 * @param array $config {
+	 *     @type string 'key'
+	 *     @type string 'version'
+	 *     @type string 'file_path'
+	 *     @type array 'data'
+	 * }
+	 *
+	 * @return void
+	 */
+	public function init_asset_data( $config ) {
+		$this->assets_config = $config;
+
+		$asset_key = $config['key'];
+
+		$asset_data = isset( $this->assets_data[ $asset_key ] ) ? $this->assets_data[ $asset_key ] : [];
+
+		if ( ! $asset_data || $this->is_asset_version_changed( $asset_data['version'] ) ) {
+			$this->set_asset_data( $asset_key );
+		}
+	}
+
+	/**
+	 * Get Asset Data.
+	 *
+	 * Getting the asset data content.
+	 *
+	 * @since 3.3.0
+	 * @access public
+	 *
+	 * @param array $config {
+	 *     @type string 'key'
+	 *     @type string 'version'
+	 *     @type string 'file_path'
+	 *     @type array 'data'
+	 * }
+	 *
+	 * @return string
+	 */
+	public function get_asset_data( $config ) {
+		$this->init_asset_data( $config );
+
+		$asset_key = $config['key'];
+
+		return $this->assets_data[ $asset_key ]['content'];
+	}
+
+	public function __construct() {
+		$assets_data = $this->get_saved_assets_data();
+
+		$content_type = $this->content_type;
+		$assets_category = $this->assets_category;
+
+		if ( ! isset( $assets_data[ $content_type ] ) ) {
+			$assets_data[ $content_type ] = [];
+		}
+
+		if ( ! isset( $assets_data[ $content_type ][ $assets_category ] ) ) {
+			$assets_data[ $content_type ][ $assets_category ] = [];
+		}
+
```

Comment:
Do this logic inside `get_saved_assets_data()`

Reply:
Done.